packages/twenty-front/src/modules/favorites/components/CurrentWorkspaceMemberFavorites.tsx

```diff
@@ -1,127 +1,229 @@
+  const [folderName, setFolderName] = useState(folder.folderName);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+
+  const { renameFolder, deleteFolder } = useFavoriteFolders();
+  const { closeDropdown, isDropdownOpen } = useDropdown(
+    `favorite-folder-${folder.folderId}`,
   );
+  const selectedFavoriteIndex = folder.favorites.findIndex((favorite) =>
+    favorite.objectNameSingular === 'view'
+      ? favorite.link === currentPathView
+      : favorite.link === currentPath,
+  );
+  const { deleteFavorite, handleReorderFavorite } = useFavorites();
+
+  const subItemArrayLength = folder.favorites.length;
+
+  const handleSubmitRename = async (value: string) => {
+    if (!value) return false;
+
+    await renameFolder(folder.folderId, value);
+    setIsRenaming(false);
+    return true;
+  };
+
+  const handleCancelRename = () => {
+    setFolderName(folder.folderName);
+    setIsRenaming(false);
+  };
+
+  const handleClickOutside = async (
+    event: MouseEvent | TouchEvent,
+    value: string,
+  ) => {
+    if (!value) {
+      setIsRenaming(false);
+      return;
+    }
 
-  if (
-    !currentWorkspaceMemberFavorites ||
-    currentWorkspaceMemberFavorites.length === 0
-  )
-    return <></>;
-
-  const isGroup = currentWorkspaceMemberFavorites.length > 1;
-
-  const draggableListContent = (
-    <DraggableList
-      onDragEnd={handleReorderFavorite}
-      draggableItems={
-        <>
-          {currentWorkspaceMemberFavorites.map((favorite, index) => {
-            const {
-              id,
-              labelIdentifier,
-              avatarUrl,
-              avatarType,
-              link,
-              recordId,
-            } = favorite;
-
-            return (
-              <DraggableItem
-                key={id}
-                draggableId={id}
-                index={index}
-                itemComponent={
-                  <StyledNavigationDrawerItem
-                    key={id}
-                    label={labelIdentifier}
-                    Icon={() => (
-                      <StyledAvatar
-                        placeholderColorSeed={recordId}
-                        avatarUrl={avatarUrl}
-                        type={avatarType}
-                        placeholder={labelIdentifier}
-                        className="fav-avatar"
-                      />
-                    )}
-                    to={link}
-                  />
-                }
-              />
-            );
-          })}
-        </>
+    await renameFolder(folder.folderId, value);
+    setIsRenaming(false);
+  };
+
+  const handleDelete = async () => {
+    if (folder.favorites.length > 0) {
+      setIsDeleteModalOpen(true);
+      closeDropdown();
+    } else {
+      await deleteFolder(folder.folderId);
+      closeDropdown();
+    }
+  };
+
+  const handleConfirmDelete = async () => {
+    await deleteFolder(folder.folderId);
+    setIsDeleteModalOpen(false);
+  };
+
+  const rightOptions = (
```

Comment:
extract this to a component, we don't create subcomponents in component to ease readability